File: client/jetbrains/src/main/java/com/sourcegraph/cody/context/RepositoryMissingEmbeddingStatus.java
package com.sourcegraph.cody.context;

import com.intellij.openapi.project.Project;
import com.sourcegraph.cody.Icons;
import com.sourcegraph.cody.config.AccountType;
import com.sourcegraph.cody.config.CodyAuthenticationManager;
import javax.swing.Icon;
import org.jetbrains.annotations.NotNull;

public class RepositoryMissingEmbeddingStatus extends RepoAvailableEmbeddingStatus {
  public RepositoryMissingEmbeddingStatus(String repoName) {
    super(repoName);
  }

  @Override
  public Icon getIcon() {
    return Icons.Repository.Missing;
  }

  @Override
  public @NotNull String getTooltip(@NotNull Project project) {
    AccountType accountType =
        CodyAuthenticationManager.getInstance().getDefaultAccountType(project);
    if (accountType == AccountType.LOCAL_APP) {
      return "Repository is not set up in Cody App";
    } else {
      return "Repository does not exist on Sourcegraph";
    }
  }
}

File: client/jetbrains/src/main/java/com/sourcegraph/cody/context/RepositoryMissingEmbeddingStatus.kt
package com.sourcegraph.cody.context

import com.intellij.openapi.project.Project
import com.sourcegraph.cody.Icons
import javax.swing.Icon

class RepositoryMissingEmbeddingStatus(val repoName: String) :
    RepoAvailableEmbeddingStatus(repoName) {

  override fun getIcon(): Icon? {
    return Icons.Repository.NoEmbedding
  }

  override fun getTooltip(project: Project): String {
    return "Repository $repoName is not indexed and has no embeddings"
  }
}
